core/class/DB.class.php

```diff
@@ -87,62 +87,37 @@ public static function &CallStoredProc($_procName, $_params, $_fetch_type, $_cla
 		}
 	}
 
-	public static function &Prepare($_query, $_params, $_fetchType = self::FETCH_TYPE_ROW, $_fetch_param = PDO::FETCH_ASSOC, $_fetch_opt = NULL) {
+	public static function &Prepare($_query, $_params, $_fetchType = self::FETCH_TYPE_ROW, $_fetch_param = PDO::FETCH_ASSOC, $_fetch_opt = null) {
+        /** @var PDOStatement $stmt */
 		$stmt = static::getConnection()->prepare($_query);
-		$res = NULL;
-		if ($stmt != false && $stmt->execute($_params) != false) {
-			if(preg_match('/^update|^replace|^delete|^create|^drop|^alter|^truncate/i', $_query)){
-				$errorInfo = $stmt->errorInfo();
-				if ($errorInfo[0] != 0000) {
-					static::$lastConnection = 0;
-					throw new Exception('[MySQL] Error code : ' . $errorInfo[0] . ' (' . $errorInfo[1] . '). ' . $errorInfo[2] . '  : ' . $_query);
-				}
-				static::$lastConnection = strtotime('now');
-				return $res;
-			}
-			if ($_fetchType == static::FETCH_TYPE_ROW) {
-				if ($_fetch_opt === null) {
-					$res = $stmt->fetch($_fetch_param);
-				} elseif ($_fetch_param == PDO::FETCH_CLASS) {
-					$res = $stmt->fetchObject($_fetch_opt);
-				}
-			} else {
-				if ($_fetch_opt === null) {
-					$res = $stmt->fetchAll($_fetch_param);
-				} else {
-					$res = $stmt->fetchAll($_fetch_param, $_fetch_opt);
-				}
-			}
-		}
-		$errorInfo = $stmt->errorInfo();
-		if ($errorInfo[0] != 0000) {
-			static::$lastConnection = 0;
-			throw new Exception('[MySQL] Error code : ' . $errorInfo[0] . ' (' . $errorInfo[1] . '). ' . $errorInfo[2] . '  : ' . $_query);
-		}
-		static::$lastConnection = strtotime('now');
-		if ($_fetch_param == PDO::FETCH_CLASS) {
-			if (is_array($res) && count($res) > 0) {
-				foreach ($res as &$obj) {
-					if (is_object($obj) && method_exists($obj, 'decrypt')) {
-						$obj->decrypt();
-						if (method_exists($obj, 'setChanged')) {
-							$obj->setChanged(false);
-						}
-					}
-				}
-			} else {
-				if (is_object($res) && method_exists($res, 'decrypt')) {
-					$res->decrypt();
-					if (method_exists($res, 'setChanged')) {
-						$res->setChanged(false);
-					}
-				}
-			}
-		}
-		return $res;
-	}
+        $stmt->execute($_params);
+
+        $errorInfo = $stmt->errorInfo();
+        if ($errorInfo[0] != 0000) {
```

Comment:
Bug to fix here ? 
```
if ($errorInfo[0] !== '00000') {
```

Quand on envoi un paramètre qui n’existe pas dans la requète, ca ne passe pas dedans.
Le vrai code 00000 (5 zéro) et est une chaine de caractère. Là on pourrait retirer 3 zéros, ca ferai la même chose.

(voir `test_query_with_bad_parameter`, qui correspond au comportement actuel, devrait probablement renvoyer une exception)